### DISCUSSION TITLE
Why only one recipe for a meal plan entry?

### DISCUSSION THREAD

**Discussion by cdesouza71** [TandoorRecipes/recipes#1261] [Q&A]
Hello,

I'm not sure that I understand the Meal-Plan feature correctly. 

When I plan a meal, I might have multiple recipes, for example a starter, two main courses and a desert. When I create a Meal-Plan entry,  I can however only choose one recipe.  

In order to plan the meal, I need to created four entries, dinner-starter, dinner-main-1, dinner-main-2, dinner-desert. This does not seem right to me.

**Comment by vabene1111** [comment]
interesting, I usually just plan multiple recipes or use sub recipes for that but you are right, having multiple recipes would be interesting as well. But there are also plans to add something like "meals" which include multiple courses and maybe optional side dishes and whatever.

**Reply by smilerz** [reply]
> interesting, I usually just plan multiple recipes or use sub recipes for that but you are right, having multiple recipes would be interesting as well. But there are also plans to add something like "meals" which include multiple courses and maybe optional side dishes and whatever.

Isn’t that what the meal plan type is intended for?

**Reply by vabene1111** [reply]
yes and no. I have meal types breakfast, lunch and dinner but a dinner could consist of a side, desert, ... but you are correct, you could also have types main course, desert, ...

I think the current solution works for now (as there are more pressing matters) but when we take a look at meals we can keep this in mind.

**Reply by cdesouza71** [reply]
Hello,

somewhere where I can enter multiple recipes for a meal would be nice instead of having to create multiple entries for a meal.

**Reply by smilerz** [reply]
> yes and no. I have meal types breakfast, lunch and dinner but a dinner could consist of a side, desert, ... but you are correct, you could also have types main course, desert, ...

those would be child meal plans, right?  lunch, dinner and breakfast can all have sides, salads, breads, whatever.  
If you have a dinner with a Main, Soup, Salad, Side, Dessert - tagging them all as a 'Dinner' MealPlan groups them all together in the calender and tags them correctly in the search view.  What is the use case for making the MealPlan <-> Recipe relationship ManyToMany?

**Reply by vabene1111** [reply]
@smilerz i am thinking of adding a model "meal" or abusing the recipe step recipes to implement something like this. I dont really want to play with the meal plan model all to much as i think it is better in this rather simple form.

**Reply by vabene1111** [reply]
@smilerz i am thinking of adding a model "meal" or abusing the recipe step recipes to implement something like this. I dont really want to play with the meal plan model all to much as i think it is better in this rather simple form. But anyone feel free to change my mind

**Reply by cdesouza71** [reply]
I think using the recipe step would not make for a good user experience as you would then 

- have two categories of recipes, basic recipes and meal recipes which would be confusing
- you would have to open the meal recipe and look at all the steps instead of just having a list of recipes
- the combinations of recipes for meals would lead to a multitude of meal recipes. Think three course meal and just changing the starter.
- finding an appropriate name for the meal recipe would be a challenge
 
but then again I'm a new user and do not have too much experience yet.

**Reply by cdesouza71** [reply]
Adding a model meal could lead to difficulties in the following areas

- you would have to open the meal to see which recipes are planned
- the combinations of recipes for meals would lead to a multitude of meals. Think three course meal and just changing the starter, this would create another meal
- finding an appropriate name for the meal would be a challenge

**Reply by cdesouza71** [reply]
Meals are more spontaneous affairs for one particular day and it should be simple to create a meal with several recipes and change out one of the recipes.

For example
- adding a starter and desert to make up the quanity of food for unexpected guests
- change out a recipe because an ingredient is missing
- change out a recipe because there is some food that needs to be eaten before it goes off
-

**Reply by vabene1111** [reply]
Very interesting write down of many arguments, thanks. I agree that meals are spontaneous so that needs to be reflected in the implementation. I will take a look at this when implementinng a solution

**Reply by Inrego** [reply]
This writeup looks promising. It's definitely something that I'm looking forward to. The current planning flow of adding multiple recipes seperately to the calendar does seem a bit off.
I have one thing to add:
While meals are indeed spontaneous, it would be nice to be able to make some pre-defined meals. Some meals in daily life are repeated often, and it would be unnecessarily cumbersome having to add all sub-recipes every time for everyday-planning.

**Reply by bradsk88** [reply]
As a quicker way to see this value without a bunch of changes to data architecture, what about just giving the user the ability to add multiple recipes for a single meal type **in the dialog**. Heck, we could even just add a "second recipe" dropdown that is optional if we want to forego the effort of building a dynamic UI.

After they click "save" in the dialog, the software would just create **two meals of that type on the same day**.

Am I oversimplifying the solution?

**Comment by Mayo-png** [comment]
I dont understand the Meal-Plan feature too:
- Already asked a [question](https://github.com/TandoorRecipes/recipes/discussions/4826) about the title
- Why can I plan a meal over several days without the number of portions being multiplied automatically? Is this intended for food that I prepare once and eat over several days? (Already asked in Discord but not really answered)
In that case, what if i prepare breakfast and want its leftovers for dinner in the next day? (Thats not possible right?)
